### PR TITLE
AI Assistant: Change image and text for interstitial based on tiers

### PR DESCRIPTION
**Pull request by dhasilva**
Part of https://github.com/Automattic/jetpack/issues/33358

## Proposed changes:

* Adds a JetpackAIInterstitialMoreRequests component for the custom last upgrade
* Sets the features and description according to the next tier
* Changes the interstitial image

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

#### On a site without the AI plan, go to My Jetpack
* Go to the purchase page, and check that the only change is the image
* Check that the purchase happens as normal and the `Manage` button is still inactive
* Check that there are no regressions on other interstitial pages

#### On local dev environment
* Sandbox your public API and change the `$current_licensed_quantity` to a known tier
* Enable the tiers on your 0-sandbox file
* Manually go to `?page=my-jetpack#/add-jetpack-ai`
* Check that the unlimited tier is shown regardless of the licensed quantity
* Change the return value of `get_next_usage_tier` to the commented code
* Check that the next tier is show

Not included: The contact button link and the checkout URLs for the tiers

### CODE REVIEW

File: projects/packages/my-jetpack/src/products/class-jetpack-ai.php [hunk 1]
```diff
@@ -74,6 +74,29 @@ public static function get_title() {
...
+			return 1;
+		}
+
+		// If the current tier is 1 or 500, there is no next tier.
+		if ( 1 === $current_tier || 500 === $current_tier ) {
+			return null;
+		}
+
+		// phpcs:ignore Squiz.PHP.CommentedOutCode.Found
+		// return $info['next-tier']['value'];
```
Comment (lhkowalski) [inline projects/packages/my-jetpack/src/products/class-jetpack-ai.php:96]: I think we are close to make this line work even without the tiers in place 🤞

File: projects/packages/my-jetpack/_inc/components/product-interstitial/index.jsx [hunk 2]
```diff
@@ -268,11 +265,26 @@ export function ExtrasInterstitial() {
...
 export function JetpackAIInterstitial() {
+	const slug = 'jetpack-ai';
+	const { detail } = useProduct( slug );
+	const { onClickGoBack } = useGoBack( { slug } );
+
+	const currentTier = detail?.[ 'ai-assistant-feature' ]?.[ 'current-tier' ]?.value;
+	const hasNextTier = ! [ 1, 500 ].includes( currentTier );
+
+	if ( ! hasNextTier ) {
+		return <JetpackAIInterstitialMoreRequests onClickGoBack={ onClickGoBack } />;
```
Comment (lhkowalski) [inline projects/packages/my-jetpack/_inc/components/product-interstitial/index.jsx:276]: In this case, are we going to show the `JetpackAIInterstitialMoreRequests` when the user has the unlimited plan?

**Comment by lhkowalski** [comment]
Code looks very good. Going to test it now.

**Review by lhkowalski — APPROVED** [review]
I tested it and it's working as expected.

When the tiers are disabled, on a free site, I get the current interstitial:

Screenshot 2023-11-10 at 11 55 04

---

On my test site, with the line commented, I always see the same interstitial no matter the tier I have on the site:

Screenshot 2023-11-10 at 12 22 41

The only different that I noted is the absense of the price on the interstitial.

---

Still on my testing site, now with the line uncommented, I can see the progression of the interstitials:

Screenshot 2023-11-10 at 12 27 41

Screenshot 2023-11-10 at 12 27 50

Screenshot 2023-11-10 at 12 28 13

Screenshot 2023-11-10 at 12 28 38

Screenshot 2023-11-10 at 12 29 03

---

With the unlimited plan active and the line uncommented, I see the same interstitial that we show for the 500 plan, asking for contact. I think the user will never land in this page on this scenario, so it's ok for me.

---

Very nice work overal, that looked like a hard problem to solve and you found a pretty neat way of doing it. Also, well done on moving the "go back" feature to a custom hook and documenting it 💯

File: projects/packages/my-jetpack/_inc/components/product-interstitial/index.jsx (hunk 2 above)
Comment (dhasilva) [reply projects/packages/my-jetpack/_inc/components/product-interstitial/index.jsx:276]: As it is, yes, but the page is not linked to once a user has a plan, so we need to think about how to migrate users.
Which reminds me that the `Manage` button should be upgraded as well.

**Pull request merged**